specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,332 @@
+                          ...layout[state.pageIndex!],
+                          label: value,
+                        })
+                      );
+                      setState({
+                        type: 'EditingState',
+                      });
+                    }
+                  : undefined
+              }
+              onAdd={
+                typeof state.pageIndex === 'number'
+                  ? undefined
+                  : (value): void => {
+                      setLayout([
+                        ...layout,
+                        {
+                          label: value,
+                          categories: [],
+                        },
+                      ]);
+                      setState({
+                        type: 'EditingState',
+                      });
+                      setActivePageIndex(layout.length);
+                    }
+              }
+              onClose={(): void => {
+                setState({ type: 'EditingState' });
+              }}
+              value={
+                typeof state.pageIndex === 'number'
+                  ? layout[state.pageIndex].label
+                  : undefined
+              }
+            />
+          )}
+          <div className="grid w-full grid-cols-[repeat(auto-fill,minmax(20rem,1fr))] gap-4 overflow-y-auto">
+            <Categories
+              onAdd={
+                isEditing
+                  ? (categoryindex): void =>
+                      typeof categoryindex === 'number'
+                        ? setState({
+                            type: 'AddingState',
+                            pageIndex: activePageIndex,
+                            categoryIndex: categoryindex,
+                          })
+                        : setLayout(
+                            replaceItem(layout, activePageIndex, {
+                              ...layout[activePageIndex],
+                              categories: [
+                                ...layout[activePageIndex].categories,
+                                {
+                                  label: '',
+                                  items: [],
+                                },
+                              ],
+                            })
+                          )
+                  : undefined
+              }
+              pageLayout={layout[activePageIndex]}
+              statsSpec={statsSpec}
+              onClick={undefined}
+              onRemove={
+                isEditing
+                  ? (categoryIndex, itemIndex): void =>
+                      setLayout(
+                        replaceItem(layout, activePageIndex, {
+                          ...layout[activePageIndex],
+                          categories:
+                            typeof itemIndex === 'number'
+                              ? replaceItem(
+                                  layout[activePageIndex].categories,
```

Comment:
Create a variable to store `layout[activePageIndex]` or even `layout[activePageIndex].categories` since you need to access them so many times